src/Diagram.js

```diff
@@ -280,9 +252,16 @@ class Diagram extends React.PureComponent {
     const { width, height } = this.containerRef.current
       ? this.containerRef.current.getBoundingClientRect()
       : DEFAULT_CONTAINER_RECT;
+    const viewport = getViewport(scroll.left, scroll.top, width, height, zoom);
+    const viewportWithOverscanning = {
+      xMin: viewport.xMin - (this.props.overscanPixels ? this.props.overscanPixels.width : width),
```

Comment:
this.props.overscanPixels this can be stored in a variable instead of doing it multiple times

Reply:
will take care of it in the next commit :)